### PR TITLE
Improve HTTP edge case handling

### PR DESCRIPTION
**Pull request by Piccirello**
-  Handle edge case where response body may not be closed 
-  Ensure HTTP connections are closed after each request 
-  Properly handle errors with non-json response

### CODE REVIEW

File: pkg/http/http.go [hunk 1]
```diff
@@ -144,6 +147,10 @@ func performRequest(req *http.Request, verifyTLS bool, params []queryParam) (int
 	requestErr := retry(5, 100*time.Millisecond, func() error {
 		resp, err := client.Do(req)
 		if err != nil {
+			if resp != nil {
+				defer response.Body.Close()
```
Comment (rgmvisser) [inline pkg/http/http.go]: `response` is still nil here, should this be `resp`?

Comment (Piccirello) [reply pkg/http/http.go]: Great catch

File: pkg/http/http.go [hunk 2]
```diff
@@ -144,6 +147,10 @@ func performRequest(req *http.Request, verifyTLS bool, params []queryParam) (int
 	requestErr := retry(5, 100*time.Millisecond, func() error {
 		resp, err := client.Do(req)
 		if err != nil {
+			if resp != nil {
+				defer resp.Body.Close()
+			}
```
Comment (rgmvisser) [inline pkg/http/http.go:152]: Small nit, but can we just take this outside of the `if err != nil {` in that case the 2nd defer is not needed anymore (and less likely to introduce a bug in the future)

**Review by rgmvisser — APPROVED** [review]

File: pkg/http/http.go (hunk 2 above)
Comment (Piccirello) [reply pkg/http/http.go:152]: I don't believe we can due to this code executing inside a function. From my understanding, once the function returns the deferred functions will execute, meaning the response body will be closed before we have a chance to read it below.

Comment (rgmvisser) [reply pkg/http/http.go:152]: Yup you are right 👍

**Pull request merged**